test/unit-test/rtcp_api/rtcp_api_utest.c

```diff
@@ -635,170 +634,1295 @@ void test_rtcpDeSerializePacket( void )
+        0xB1, 0xB2, 0xB3, 0xB4, /* Inter-arrival Jitter = 0xB1B2B3B4. */
+        0xC1, 0xC2, 0xC3, 0xC4, /* Last SR = 0xC1C2C3C4. */
+        0x5A, 0x5B, 0x5C, 0x5D, /* Delay since last SR = 0x5A5B5C5D. */
+        /* Reception Report 2. */
+        0x00, 0x00, 0x00, 0x02, /* SSRC of second source. */
+        0x11, 0xA0, 0xA1, 0xA2, /* Fraction lost = 0x11, Cumulative packet lost = 0xA0A1A2. */
+        0xD1, 0xD2, 0xD3, 0xD4, /* Extended highest sequence number received = 0xD1D2D3D4. */
+        0xB1, 0xB2, 0xB3, 0xB4, /* Inter-arrival Jitter = 0xB1B2B3B4. */
+        0xC1, 0xC2, 0xC3, 0xC4, /* Last SR = 0xC1C2C3C4. */
+        0x5A, 0x5B, 0x5C, 0x5D, /* Delay since last SR = 0x5A5B5C5D. */
+    };
+    size_t serializedPacketLength = sizeof( serializedPacket );
     RtcpPacket_t rtcpPacket = { 0 };
-    RtcpFirPacket_t rtcpFirPacket;
     RtcpResult_t result;
-    uint8_t payloadBuffer[ 6 ];
 
-    result = Rtcp_ParseFirPacket( NULL,
-                                  &( rtcpPacket ),
-                                  &( rtcpFirPacket ) );
-
-    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
-                       result );
-
-    result = Rtcp_ParseFirPacket( &( context ),
-                                  NULL,
-                                  &( rtcpFirPacket ) );
+    result = Rtcp_Init( &( context ) );
 
-    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+    TEST_ASSERT_EQUAL( RTCP_RESULT_OK,
                        result );
 
-    result = Rtcp_ParseFirPacket( &( context ),
-                                  &( rtcpPacket ),
-                                  NULL );
+    result = Rtcp_DeserializePacket( &( context ),
+                                     &( serializedPacket[ 0 ] ),
+                                     serializedPacketLength,
+                                     &( rtcpPacket ) );
 
-    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
+    TEST_ASSERT_EQUAL( RTCP_RESULT_OK,
                        result );
+    TEST_ASSERT_EQUAL( 0,
+                       rtcpPacket.header.padding );
+    TEST_ASSERT_EQUAL( 0,
+                       rtcpPacket.header.receptionReportCount );
+    TEST_ASSERT_EQUAL_PTR( &( serializedPacket[ 4 ] ),
+                           rtcpPacket.pPayload );
+    TEST_ASSERT_EQUAL( serializedPacketLength - 4,
+                       rtcpPacket.payloadLength );
+}
 
-    rtcpPacket.pPayload = NULL;
-
-    result = Rtcp_ParseFirPacket( &( context ),
-                                  &( rtcpPacket ),
-                                  &( rtcpFirPacket ) );
-
-    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
-                       result );
+/*-----------------------------------------------------------*/
 
-    /* Payload length less than RTCP_FIR_PACKET_PAYLOAD_LENGTH. */
-    rtcpPacket.payloadLength = 2;
-    rtcpPacket.pPayload = &( payloadBuffer[ 0 ] );
+/**
+ * @brief Validate RTCP DeSerialize Packet functionality.
```

Comment:
What part of this test case is invalid? The result seems normal in this test case.

Reply:
The test is for invalid case cause for a Feedback of type FIR, the Packet type should be 192 and the FMT (Feedback Message type ) should be 0 , here its 2 .

Reply:
I wrote a comment on the like 698 regarding it. Please Let me know if its unclear or wording needed to be improved.